stonesoup/types/orbitalstate.py

```diff
@@ -0,0 +1,840 @@
+    conversions are generally found in, or derived from, [1].
+
+    The object is constructed from the input vector :math:`X_{t_{0}}`
+    at epoch :attr:`State.timestamp` :math:`t_0` in the appropriate
+    coordinates indicated via the keyword:
+
+        coordinates = "Cartesian" (the orbital state vector),
+
+             .. math::
+
+                X_{t_0} = [r_x, r_y, r_z, \dot{r}_x, \dot{r}_y,
+                    \dot{r}_z]^{T}
+
+            where :math:`r_x, r_y, r_z` are the Cartesian position
+            coordinates in the Primary-Centered Inertial frame and
+            :math:`\dot{r}_x, \dot{r}_y, \dot{r}_z` are the
+            corresponding velocity coordinates.
+
+        coordinates = "Keplerian" (Keplarian elements),
+
+            .. math::
+
+                X_{t_0} = [e, a, i, \Omega, \omega, \theta]^{T}
+
+            where :math:`e` is the orbital eccentricity (unitless),
+            :math:`a` the semi-major axis ([length]), :math:`i` the
+            inclination (radian), :math:`\Omega` is the longitude of the
+            ascending node (radian), :math:`\omega` the argument of
+            periapsis (radian), and :math:`\theta` the true anomaly
+            (radian).
+
+        coordinates = "TLE" (Two-Line elements, [2]),
+
+            .. math::
+
+                X_{t_0} = [i, \Omega, e, \omega, M_0, n]^{T}
+
+            where :math:`i` the inclination (radians), :math:`\Omega`
+            is the longitude of the ascending node (radian), :math:`e`
+            is the orbital eccentricity (unitless), :math:`\omega` the
+            argument of perigee (radian), :math:`M_0` the mean anomaly
+            (radian) :math:`n` the mean motion (radian
+            [time] :math:`^{-1}`)
+
+        coordinates = "Equinoctial" (equinoctial elements, [3])
+
+            .. math::
+
+                X_{t_0} = [a, h, k, p, q, \lambda]^{T}
+
+            where :math:`a` the semi-major axis ([length]),
+            :math:`h` is the horizontal component of the
+            eccentricity, :math:`k` is the vertical component of the
+            eccentricity, :math:`p` is the horizontal component of the
+            inclination, :math:`q` is the vertical component of the
+            inclination and :math:`\lambda` is the mean longitude
+
+    The gravitational parameter :math:`\mu = GM` can be defined. If
+    left undefined it defaults to that of the Earth, :math:`3.986004418
+    \, (\pm \, 0.000000008) \times 10^{14} \mathrm{m}^3 \mathrm{s}^{−2}`
+
+    References
+    ----------
+    1. Curtis, H.D. 2010, Orbital Mechanics for Engineering
+    Students (3rd Ed), Elsevier Aerospace Engineering Series
+
+    2. https://spaceflight.nasa.gov/realdata/sightings/SSapplications/\
+    Post/JavaSSOP/SSOP_Help/tle_def.html
+
+    3. Broucke, R. A. & Cefola, P. J. 1972, Celestial Mechanics, Volume
+    5, Issue 3, pp. 303-310
+
+    """
+
+    coordinates = Property(
+        str, default='Cartesian',
```

Comment:
Using a [`Enum`](https://docs.python.org/3/library/enum.html) rather than a string would be better for this property.

Reply:
I don't quite get it. `Enum` is a (restricted) set; `coordinates` has to be a member of that set. But then I'd need to create the `Enum` somewhere (where?) and run exactly the same logic over its members. 

?

Reply:
Yeah, it makes it more explicit what values are allowed, versus a string. You could create an Enum in this file. You can also have [descriptive strings](https://docs.python.org/3/library/enum.html#using-a-descriptive-string), rather than numbers, which may be better for this use case.

Reply:
But then I can't initialise via state = OrbitalState(statevector, coordinates="Keplerian") anymore. I now have to have the wherewithal to create some `CoordinateSystem(Enum)`, say `enumerated_coordinates =  CoordinateSystem("Keplerian")` and then pass that as the keyword. Just telling a user which string to use seems easier.

Reply:
But you can do: `OrbitalState(statevector, coordinates=CoordinateSystem.KEPLERIAN)`, which will support autocompletion.
Also `CoordinateSystem("Keplerian") is  CoordinateSystem(CoordinateSystem("Keplerian"))`, so can also use it to support strings still, as well as provide input validation:
```python
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.coordinates = CoordinateSystem(self.coordinates)
```

Reply:
What if I want to do coordinates="keplerian", or coordinates="kEplEriAN"?

Reply:
> What if I want to do coordinates="keplerian", or coordinates="kEplEriAN"?

Yes, another good reason why clearly defined constants/enums are preferable vs. strings.

Reply:
I'd argue the opposite. At present coordinates="Keplerian" or coordinates="keplerian" both work. Your proposal would make one of them fail, => less flexibility.

Reply:
Where a defined supported values are available, enums (or string constants) should really be used. This way they are defined in code, as well as documentation. Whilst we should encourage use of enum value, you could still have case insensitive string support by using Enum `_missing_` class method:
```python
    @classmethod
    def _missing_(cls, value):
        for element in cls:
            if element.value.lower() == value.lower():
                return element
```